src/Yoti/Entity/DocumentDetails.php

```diff
@@ -0,0 +1,181 @@
+<?php
+namespace Yoti\Entity;
+
+use Yoti\Exception\AttributeException;
+
+class DocumentDetails
+{
+    /**
+     * The values of the Document Details are in the format and order as defined in this pattern
+     * e.g PASS_CARD GBR 22719564893 - CITIZENCARD, the last two are optionals
+     */
+    const VALIDATION_PATTERN = '/^([A-Za-z_]*) ([A-Za-z]{3}) ([A-Za-z0-9]{1}).*$/';
+    const TYPE_INDEX = 0;
+    const COUNTRY_INDEX = 1;
+    const NUMBER_INDEX = 2;
+    const EXPIRATION_INDEX = 3;
+    const AUTHORITY_INDEX = 4;
+
+    /**
+     * @var string
+     */
+    private $type;
+
+    /**
+     * @var string
+     */
+    private $issuingCountry;
+
+    /**
+     * @var \DateTime
+     */
+    private $expirationDate;
+
+    /**
+     * @var string
+     */
+    private $documentNumber;
+
+    /**
+     * @var string
+     */
+    private $issuingAuthority;
+
+    /**
+     * @var array
+     */
+    private $parsedValues = [];
```

Comment:
Do we need this here, or can it just be local to the _parseFromValue()_ method?

Reply:
Yes it can, but I will have to pass the parsedValues as a param to each setter in that case.

Reply:
Probably cleaner to do that.  Makes the behaviour more obvious.